src/libraries/Microsoft.PowerFx.Core/Functions/TexlFunction.cs

```diff
@@ -1080,6 +1086,26 @@ protected bool CheckBooleanColumnType(DType type, TexlNode arg, IErrorContainer
             return CheckColumnType(type, arg, DType.Boolean, errors, TexlStrings.ErrInvalidSchemaNeedColorCol_Col, ref nodeToCoercedTypeMap);
         }
 
+        protected DType NumDecReturnType(CheckTypesContext context, DType argType)
+        {
+            IEnumerable<TypedName> columns;
+
+            if (argType.IsTable && (columns = argType.GetNames(DPath.Root)).Count() == 1)

```

Comment:
>columns = argType.GetNames(DPath.Root) [](http://example.com/codeflow?start=36&length=38)

is the assignment necessary? the added complexity makes the condition a bit hard to read, either pulling the assignment before the condition or just repeating the GetNames call would be better